server/src/services/search.service.ts

```diff
@@ -144,6 +156,97 @@ export class SearchService {
     }
   }
 
+  async handleQueueSearchDuplicates({ force }: IBaseJob): Promise<JobStatus> {
+    const { machineLearning } = await this.configCore.getConfig();
+    if (!machineLearning.enabled || !machineLearning.clip.enabled) {
+      return JobStatus.SKIPPED;
+    }
+
+    const assetPagination = usePagination(JOBS_ASSET_PAGINATION_SIZE, (pagination) => {
+      return force
+        ? this.assetRepository.getAll(pagination, { isVisible: true })
+        : this.assetRepository.getWithout(pagination, WithoutProperty.DUPLICATE);
+    });
+
+    for await (const assets of assetPagination) {
+      await this.jobRepository.queueAll(
+        assets.map((asset) => ({ name: JobName.DUPLICATE_DETECTION, data: { id: asset.id } })),
+      );
+    }
+
+    return JobStatus.SUCCESS;
+  }
+
+  async handleSearchDuplicates({ id }: IEntityJob): Promise<JobStatus> {
+    const { machineLearning } = await this.configCore.getConfig();
+    if (!machineLearning.enabled || !machineLearning.duplicateDetection.enabled) {
```

Comment:
Do you think it would be worth introducing a `DUPLICATE_DETECTION` feature flag?

Reply:
I added a flag in my test branch. I think we should and I think mertalev agrees with it.

Reply:
```suggestion
    if (await this.configCore.hasFeature(FeatureFlag.DUPLICATE_DETECTION)) {
```
Still need to add the flags (I already did and tested it on a fork.

Reply:
I think this is also wrong, I think it needs to be inverted.

Reply:
Is the purpose of the changes to this file so that you can search for duplicate files?

Reply:
I think it's for searching for the Duplicate?

Reply:
Figured it out.